source/fe/fe.cc

```diff
@@ -59,8 +59,7 @@ FiniteElement<dim, spacedim>::FiniteElement(
   const std::vector<bool> &         r_i_a_f,
   const std::vector<ComponentMask> &nonzero_c)
   : FiniteElementData<dim>(fe_data)
-  , adjust_line_dof_index_for_line_orientation_table(
-      dim == 3 ? this->n_dofs_per_line() : 0)
+  , adjust_line_dof_index_for_line_orientation_table(this->n_dofs_per_line())
```

Comment:
We didn't need this in the 2d case before, though I don't understand why. Can you explain why that wasn't necessary but now is?

Reply:
Would you like me to add a comment here, or in general? I added some extra text circa line 573 (where this array is actually used).

The short answer is that all-quad grids should be correctly oriented but this isn't always true any more due to periodic boundary conditions. IIRC doing uniform triangle refinement results in an inconsistently oriented grid so we now need to use this array in 2D.

Reply:
No need for to add anything to the code/comments. Thanks for the clarification!